src/contributors/02-local-development.md

```diff
@@ -105,7 +108,7 @@ To run federation/API tests, first add the following lines to `/etc/hosts`:
 127.0.0.1       lemmy-epsilon
 ```
 
-Then run the following script:
+You will need to hand-edit the api_tests/run-federation-test.sh script file, add your PostgreSQL lemmy password. Note, the LEMMY_DATABASE_URL environment variable inside that script file is not in the same format as normally required, the "/lemmy" database is not on the URL. Then run the following script:
```

Comment:
This is not a good solution because it will result in conflicts or accidentally committing those changes.

Reply:
You do see the issue, it being hard-coded in the script? https://github.com/LemmyNet/lemmy/blob/1f21bdb2f920fc638b8581bb1a89af0c066e0874/api_tests/run-federation-test.sh#L4

As a solution,  I suggest some code be added to derive the needed environment variable from the existing one, removing the "/lemmy" at the end using bash code. That way the password can be kept out of the file.

Reply:
Using a default password or DB isn't an issue for local development, only for prod setups.

Reply:
You could also change it to something like this to read the db url from environment, or fallback to hardcoded one otherwise.

https://stackoverflow.com/a/2013589